server/lib/matchmaking/matchmaking-api.ts

```diff
@@ -134,19 +139,27 @@ export class MatchmakingApi {
   }
 
   @httpDelete('/find')
-  @httpBefore(throttleMiddleware(matchmakingThrottle, ctx => String(ctx.session!.user!.id)))
+  @httpBefore(
+    ensureLoggedIn,
+    throttleMiddleware(matchmakingThrottle, ctx => String(ctx.session!.user!.id)),
+  )
   async cancelSearch(ctx: RouterContext): Promise<void> {
     await this.matchmakingService.cancel(ctx.session!.user!.id)
   }
 
   @httpPost('/accept')
-  @httpBefore(throttleMiddleware(matchmakingThrottle, ctx => String(ctx.session!.user!.id)))
+  @httpBefore(
+    ensureLoggedIn,
+    throttleMiddleware(matchmakingThrottle, ctx => String(ctx.session!.user!.id)),
+  )
   async acceptMatch(ctx: RouterContext): Promise<void> {
     await this.matchmakingService.accept(ctx.session!.user!.id)
   }
 
   @httpGet('/seasons')
-  @httpBefore(checkAllPermissions('manageMatchmakingSeasons'))
+  @httpBefore(
+    throttleMiddleware(seasonsRetrievalThrottle, ctx => String(ctx.session?.user?.id) ?? ctx.ip),
```

Comment:
can you file an issue to pull these throttle identifier functions out into something common? Copy-pasting these everywhere is not great and it's not entirely clear what their intention is without deeper thought